sdk/core/azure-core/azure/core/utils/_pipeline_transport_rest_shared.py

```diff
@@ -344,16 +345,31 @@ def _format_data_helper(data: Union[str, IO]) -> Union[Tuple[None, str], Tuple[O
     :rtype: tuple[str, IO, str] or tuple[None, str]
     :return: A tuple of (data name, data IO, "application/octet-stream") or (None, data str)
     """
-    if hasattr(data, "read"):
+    content_type = "application/octet-stream"  # default content type
+    filename: Optional[str] = None
+    if isinstance(data, tuple):
+        if len(data) == 2:
+            # Filename and file bytes are included
+            filename, file_bytes = cast(Tuple[Optional[str], FileContent], data)
+        elif len(data) == 3:
+            # Filename, file object, and content_type are included
+            filename, file_bytes, content_type = cast(Tuple[Optional[str], FileContent, str], data)
+        else:
+            raise ValueError(
+                "Unexpected data format. Expected file, or tuple of (filename, file_bytes) or "
+                "(filename, file_bytes, content_type)."
+            )
+    else:
+        # here we just get the file content
         data = cast(IO, data)
-        data_name = None
         try:
             if data.name[0] != "<" and data.name[-1] != ">":
```

Comment:
How often does an `IO` have a `name` Attribute?

Reply:
This code already existed, but I believe if it's a readable IO, we usually can get the filename